tests/cache/scalar.rs

```diff
@@ -208,18 +214,1121 @@ mod increment {
     }
 }
 
-mod set_if_not_exists {}
+mod set_if_absent {
+    use super::*;
+
+    #[tokio::test]
+    async fn invalid_cache_name() -> MomentoResult<()> {
+        let client = &CACHE_TEST_STATE.client;
+        let result = client
+            .set_if_absent("   ", "key", "value")
+            .await
+            .unwrap_err();
+        assert_eq!(result.error_code, MomentoErrorCode::InvalidArgumentError);
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn nonexistent_cache() -> MomentoResult<()> {
+        let client = &CACHE_TEST_STATE.client;
+        let cache_name = unique_string("fake-cache");
+        let result = client
+            .set_if_absent(cache_name, "key", "value")
+            .await
+            .unwrap_err();
+        assert_eq!(result.error_code, MomentoErrorCode::NotFoundError);
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn string_key_string_value() -> MomentoResult<()> {
+        let client = &CACHE_TEST_STATE.client;
+        let cache_name = CACHE_TEST_STATE.cache_name.as_str();
+        let item1 = TestScalar::new();
+        let item2 = TestScalar::new();
+
+        // Setting a key that doesn't exist should create it
+        let result = client
+            .set_if_absent(cache_name, item1.key(), item1.value())
+            .await?;
+        assert_eq!(result, SetIfAbsent::Stored);
+        let result_value: String = client.get(cache_name, item1.key()).await?.try_into()?;
```

Comment:
This is a good pattern. We can use this or the `Get::from` that I demonstrated. For this I recommend adding a message if the `try_into` fails:

```suggestion
        let result = client.get(cache_name, item1.key()).await?;
        let result_value: String = result.try_into().expect("Expected Get::Hit but got {?:}", result);
```

That way our integration tests can have something to say about what the value was whenever the `try_into` fails.

Reply:
I like the `Get::from` approach a lot, and the `assert_eq!` should handle showing the diff if the result isn't as expected, so I'll switch similar instances to `Get::from`